R/summarise_scores.R

```diff
@@ -0,0 +1,112 @@
+    filter(forecast_date <= last_forecast_date,
+           target_end_date <= report_date)
+
+  ## duplicate country data as overall data
+  score_df <- score_data %>%
+    mutate(location = "Overall") %>%
+    bind_rows(score_data)
+
+  num_loc <- score_df %>%
+    group_by(model, location, target_variable, horizon) %>%
+    summarise(n_loc = n_distinct(location_name), .groups = "drop")
+
+  ## for overall, if more than 1 location exists, filter to have at least half
+  ## of them
+  score_df <- score_df %>%
+    group_by(model, target_variable, location, horizon) %>%
+    mutate(n = n_distinct(location_name)) %>%
+    ungroup() %>%
+    mutate(nall = n_distinct(location_name)) %>%
+    filter(location != "Overall" | n >= nall / 2) %>%
+    select(-n, -nall)
+
+  ## continuous weeks of submission
+  if (restrict_weeks > 0) {
+    cont_weeks <- score_df %>%
+      group_by(model, location, target_variable, target_end_date, horizon, forecast_date) %>%
+      summarise(present = 1, .groups = "drop") %>%
+      complete(model, location, target_variable, target_end_date, horizon) %>%
+      filter(forecast_date <= report_date - weeks(horizon)) %>%
+      group_by(model, location, target_variable, horizon) %>%
+      mutate(continuous_weeks = cumsum(rev(present))) %>%
+      filter(!is.na(continuous_weeks)) %>%
+      summarise(continuous_weeks = max(continuous_weeks), .groups = "drop")
+
+    score_df <- score_df %>%
+      left_join(cont_weeks, by = c(
+                              "model", "target_variable", "horizon",
+                              "location"
+                            )) %>%
+      replace_na(list(continuous_weeks = 0)) %>%
+      filter(continuous_weeks >= restrict_weeks) %>%
+      select(-continuous_weeks)
+  }
+
+  ## number of forecasts
+  num_fc <- score_df %>%
+    count(model, target_variable, horizon, location)
+
+  rel_ae <- score_df %>%
+    select(model, target_variable, horizon, location, aem) %>%
+    pairwise_comparison(
+      metric = "aem",
+      baseline = "EuroCOVIDhub-baseline",
+      summarise_by = c("model", "target_variable", "horizon", "location")
+    ) %>%
+    select(model, target_variable, horizon, location,
+           rel_ae = scaled_rel_skill) %>%
+    distinct()
+
+  rel_wis <- score_df %>%
+    filter(n_quantiles == 23) %>%
+    select(model, target_variable, horizon, location, interval_score = wis) %>%
+    pairwise_comparison(
+      metric = "interval_score",
+      baseline = "EuroCOVIDhub-baseline",
+      summarise_by = c("model", "target_variable", "horizon", "location")
+    ) %>%
+    select(model, target_variable, horizon, location, rel_wis = scaled_rel_skill) %>%
+    distinct()
+
+  ## calibration metrics (50 and 95 percent coverage and bias)
+  coverage <- score_df %>%
+    group_by(model, target_variable, horizon, location) %>%
+    summarise(across(starts_with("cov_"), mean), .groups = "drop")
+  bias <- score_df %>%
+    filter(n_quantiles == 23) %>%
```

Comment:
We used to compute bias even if not all quantiles were present.

https://github.com/epiforecasts/EuroForecastHub/blob/6f2dfd7dc6906c7d806ed343aed6b4731f197ae1/R/score_models.R#L92-L99

Now, we require the presence of the 23 quantiles to compute it. Was the previous approach wrong?

It seems so because it wasn't doing what it should have done according to the comment a dozen lines later:

https://github.com/epiforecasts/EuroForecastHub/blob/6f2dfd7dc6906c7d806ed343aed6b4731f197ae1/R/score_models.R#L117

Reply:
Yes, I think it was previously wrong - the values are not comparable between models if the underlying quantiles are different.